### ISSUE TITLE
Can't create Backup in version 2.1.26

### ISSUE THREAD

**Issue by nexosone** [nilsteampassnet/TeamPass#1566]
### Steps to reproduce
1. Start Backup Script

### Expected behaviour

Create a valid Backup on server

### Actual behaviour

PHP Error

### Server configuration
**Operating system**:
Debia 8.6

**Web server:**
Server version: Apache/2.4.10 (Debian)

**Database:**
mysqld  Ver 5.5.52-0+deb8u1 for debian-linux-gnu on x86_64 (

**PHP version:**
PHP 5.6.27-0+deb8u1 (cli) (built: Oct 15 2016 15:53:28)
Copyright (c) 1997-2016 The PHP Group
Zend Engine v2.6.0, Copyright (c) 1998-2016 Zend Technologies
    with Zend OPcache v7.0.6-dev, Copyright (c) 1999-2016, by Zend Technologies
**Teampass version:** 

2.1.26
Updated from 2.1.25 Teampass

### Client configuration
**Browser:**
equal
**Operating system:**
equal
#### Web server error log
```
[Tue Nov 01 12:18:40.651932 2016] [:error] [pid 19873] [client <insert-fancy-IP-here>:61874] PHP Warning:  include(../includes/settings.php): failed to open stream: No such file or directory in <insert-fancy-path-here>/teampass/backups/script.backup.php on line 15, referer: https://<insert-fancy-adress-here>/teampass/backups/
[Tue Nov 01 12:18:40.652197 2016] [:error] [pid 19873] [client <insert-fancy-IP-here>:61874] PHP Warning:  include(): Failed opening '../includes/settings.php' for inclusion (include_path='.:/usr/share/php:/usr/share/pear') in <insert-fancy-path-here>/teampass/backups/script.backup.php on line 15, referer: https://<insert-fancy-adress-here>/teampass/backups/
[Tue Nov 01 12:18:40.652701 2016] [:error] [pid 19873] [client <insert-fancy-IP-here>:61874] PHP Notice:  Undefined variable: server in <insert-fancy-path-here>/teampass/backups/script.backup.php on line 22, referer: https://<insert-fancy-adress-here>/teampass/backups/
[Tue Nov 01 12:18:40.652790 2016] [:error] [pid 19873] [client <insert-fancy-IP-here>:61874] PHP Notice:  Undefined variable: user in <insert-fancy-path-here>/teampass/backups/script.backup.php on line 23, referer: https://<insert-fancy-adress-here>/teampass/backups/
[Tue Nov 01 12:18:40.652872 2016] [:error] [pid 19873] [client <insert-fancy-IP-here>:61874] PHP Notice:  Undefined variable: pass in <insert-fancy-path-here>/teampass/backups/script.backup.php on line 24, referer: https://<insert-fancy-adress-here>/teampass/backups/
[Tue Nov 01 12:18:40.652956 2016] [:error] [pid 19873] [client <insert-fancy-IP-here>:61874] PHP Notice:  Undefined variable: database in <insert-fancy-path-here>/teampass/backups/script.backup.php on line 25, referer: https://<insert-fancy-adress-here>/teampass/backups/
[Tue Nov 01 12:18:40.653031 2016] [:error] [pid 19873] [client <insert-fancy-IP-here>:61874] PHP Notice:  Undefined variable: port in <insert-fancy-path-here>/teampass/backups/script.backup.php on line 26, referer: https://<insert-fancy-adress-here>/teampass/backups/
[Tue Nov 01 12:18:40.653107 2016] [:error] [pid 19873] [client <insert-fancy-IP-here>:61874] PHP Notice:  Undefined variable: server in <insert-fancy-path-here>/teampass/backups/script.backup.php on line 28, referer: https://<insert-fancy-adress-here>/teampass/backups/
[Tue Nov 01 12:18:40.653217 2016] [:error] [pid 19873] [client <insert-fancy-IP-here>:61874] PHP Notice:  Undefined variable: user in <insert-fancy-path-here>/teampass/backups/script.backup.php on line 28, referer: https://<insert-fancy-adress-here>/teampass/backups/
[Tue Nov 01 12:18:40.653289 2016] [:error] [pid 19873] [client <insert-fancy-IP-here>:61874] PHP Notice:  Undefined variable: pass in <insert-fancy-path-here>/teampass/backups/script.backup.php on line 28, referer: https://<insert-fancy-adress-here>/teampass/backups/
[Tue Nov 01 12:18:40.653403 2016] [:error] [pid 19873] [client <insert-fancy-IP-here>:61874] PHP Notice:  Undefined variable: database in <insert-fancy-path-here>/teampass/backups/script.backup.php on line 28, referer: https://<insert-fancy-adress-here>/teampass/backups/
[Tue Nov 01 12:18:40.653474 2016] [:error] [pid 19873] [client <insert-fancy-IP-here>:61874] PHP Notice:  Undefined variable: port in <insert-fancy-path-here>/teampass/backups/script.backup.php on line 28, referer: https://<insert-fancy-adress-here>/teampass/backups/
[Tue Nov 01 12:18:40.654070 2016] [:error] [pid 19873] [client <insert-fancy-IP-here>:61874] PHP Warning:  mysqli_connect(): (HY000/1045): Access denied for user ''@'localhost' (using password: NO) in <insert-fancy-path-here>/teampass/backups/script.backup.php on line 28, referer: https://<insert-fancy-adress-here>/teampass/backups/
[Tue Nov 01 12:18:40.654252 2016] [:error] [pid 19873] [client <insert-fancy-IP-here>:61874] PHP Notice:  Undefined variable: pre in <insert-fancy-path-here>/teampass/backups/script.backup.php on line 35, referer: https://<insert-fancy-adress-here>/teampass/backups/
[Tue Nov 01 12:18:40.654681 2016] [:error] [pid 19873] [client <insert-fancy-IP-here>:61874] PHP Warning:  mysqli::mysqli(): (HY000/1045): Access denied for user ''@'localhost' (using password: NO) in <insert-fancy-path-here>/teampass/includes/libraries/Database/Meekrodb/db.class.php on line 163, referer: https://<insert-fancy-adress-here>/teampass/backups/
[Tue Nov 01 12:18:40.654835 2016] [:error] [pid 19873] [client <insert-fancy-IP-here>:61874] PHP Fatal error:  Call to undefined function prepareExchangedData() in <insert-fancy-path-here>/teampass/includes/libraries/Database/Meekrodb/db.class.php on line 925, referer: https://<insert-fancy-adress-here>/teampass/backups/
```

**Comment by frankenstein91** [comment]
same problem on ubuntu server

**Comment by nilsteampassnet** [comment]
Tested on Windows server, no problem.

The bck file is correctly generated
![image](https://cloud.githubusercontent.com/assets/1197546/20039979/e1d303ee-a44e-11e6-9f75-2816b7f74a8f.png)

Are you sure you have updated all the files to 2.1.26?
When looking at the error, I can see that it refers to settings.php file that is searched in `includes` folder. But since 2.1.26, the file is in `includes/config/` and the script has been changed.

So you should upgrade to 2.1.26 the files in "backups" too.

**Comment by pkoevesdi** [comment]
Same here: Have Teampass running on Windows XAMPP. 
From the Backup Script I get back:

```
Warning: unlink(C:\xampp\htdocs\backups/bck_teampass-1511780516.sql): No such file or directory in C:\xampp\htdocs\backups\script.backup.php on line 113

Warning: rename(C:\xampp\htdocs\backups/bck_teampass-1511780516.encrypted.sql,C:\xampp\htdocs\backups/bck_teampass-1511780516.sql): Das System kann die angegebene Datei nicht finden. (code: 2) in C:\xampp\htdocs\backups\script.backup.php on line 117
[{"error":"" , "status":"success"}]
```

I tried to play around with the backup path in Teampass admin Page (tried different directory separators and Salsh in the end) - no luck. Then I realized that the above error message contains _different_ separator before the filename.
So I changed in lines 107 and following the composition of the paths to:
join(DIRECTORY_SEPARATOR,array($settings['bck_script_path'],$bck_filename.'.sql'))

Now the error message gets at least consistent path separators, but still not working. I suppose, where I started was the file renaming and deleting part. I think, the "join"-command must be used at the file creation part too, but I didn't find that. This is, where I stop my debugging.
